src/OVAL/probes/probe-table.c

```diff
@@ -77,45 +193,123 @@ typedef struct probe_table_entry {
+#endif
+#ifdef LINUX_RPMINFO_PROBE
 	{OVAL_LINUX_RPM_INFO, rpminfo_probe_init, rpminfo_probe_main, rpminfo_probe_fini, rpminfo_probe_offline_mode},
+#endif
+#ifdef LINUX_RPMVERIFY_PROBE
 	{OVAL_LINUX_RPMVERIFY, rpmverify_probe_init, rpmverify_probe_main, rpmverify_probe_fini, rpmverify_probe_offline_mode},
+#endif
+#ifdef LINUX_RPMVERIFYFILE_PROBE
 	{OVAL_LINUX_RPMVERIFYFILE, rpmverifyfile_probe_init, rpmverifyfile_probe_main, rpmverifyfile_probe_fini, rpmverifyfile_probe_offline_mode},
+#endif
+#ifdef LINUX_RPMVERIFYPACKAGE_PROBE
 	{OVAL_LINUX_RPMVERIFYPACKAGE, rpmverifypackage_probe_init, rpmverifypackage_probe_main, rpmverifypackage_probe_fini, rpmverifypackage_probe_offline_mode},
+#endif
+#ifdef LINUX_SELINUXBOOLEAN_PROBE
 	{OVAL_LINUX_SELINUXBOOLEAN, NULL, selinuxboolean_probe_main, NULL, NULL},
+#endif
+#ifdef LINUX_SELINUXSECURITYCONTEXT_PROBE
 	{OVAL_LINUX_SELINUXSECURITYCONTEXT, NULL, selinuxsecuritycontext_probe_main, NULL, NULL},
+#endif
+#ifdef LINUX_SYSTEMDUNITDEPENDENCY_PROBE
 	{OVAL_LINUX_SYSTEMDUNITDEPENDENCY, NULL, systemdunitdependency_probe_main, NULL, NULL},
+#endif
+#ifdef LINUX_SYSTEMDUNITPROPERTY_PROBE
 	{OVAL_LINUX_SYSTEMDUNITPROPERTY, NULL, systemdunitproperty_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_DNSCACHE_PROBE
 	{OVAL_UNIX_DNSCACHE, NULL, dnscache_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_FILE_PROBE
 	{OVAL_UNIX_FILE, file_probe_init, file_probe_main, file_probe_fini, NULL},
+#endif
+#ifdef UNIX_FILEEXTENDEDATTRIBUTE_PROBE
 	{OVAL_UNIX_FILEEXTENDEDATTRIBUTE, fileextendedattribute_probe_init, fileextendedattribute_probe_main, fileextendedattribute_probe_fini, NULL},
+#endif
+#ifdef UNIX_GCONF_PROBE
 	{OVAL_UNIX_GCONF, NULL, gconf_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_INTERFACE_PROBE
 	{OVAL_UNIX_INTERFACE, NULL, interface_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_PASSWORD_PROBE
 	{OVAL_UNIX_PASSWORD, password_probe_init, password_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_PROCESS_PROBE
 	{OVAL_UNIX_PROCESS, NULL, process_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_PROCESS58_PROBE
 	{OVAL_UNIX_PROCESS58, NULL, process58_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_ROUTINGTABLE_PROBE
 	{OVAL_UNIX_ROUTINGTABLE, NULL, routingtable_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_RUNLEVEL_PROBE
 	{OVAL_UNIX_RUNLEVEL, runlevel_probe_init, runlevel_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_SHADOW_PROBE
 	{OVAL_UNIX_SHADOW, shadow_probe_init, shadow_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_SYMLINK_PROBE
 	{OVAL_UNIX_SYMLINK, symlink_probe_init, symlink_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_SYSCTL_PROBE
 	{OVAL_UNIX_SYSCTL, NULL, sysctl_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_UNAME_PROBE
 	{OVAL_UNIX_UNAME, NULL, uname_probe_main, NULL, NULL},
+#endif
+#ifdef UNIX_XINETD_PROBE
 	{OVAL_UNIX_XINETD, xinetd_probe_init, xinetd_probe_main, xinetd_probe_fini, NULL},
-	{OVAL_INDEPENDENT_SYSCHAR_SUBTYPE, system_info_probe_init, system_info_probe_main, NULL, NULL},
+#endif
```

Comment:
having to deal with each probe twice in this file is a pain, can't this be done more elegantly?

Reply:
The problem is that to add a function name to this table we need to include a header file in which the function is defined. I thought about having a new header file in which we  include other header files, but then you would have to edit 2 files instead of 1 twice. Which is better?

Reply:
ok nvm. There probably is a solution where you can do both under one #ifdef but I don't know it off the top of my head.

Reply:
Maybe we could have a macro that expands to `{OVAL_*SUBTYPE, *probe_init, *probe_main ...}` if something is defined and to an empty string if ti isn't defined, and the use it in this structure.